pkg/pipelines/webhook/webhook.go

```diff
@@ -97,17 +104,51 @@ func newWebhookInfo(accessToken, pipelinesFile string, serviceName *QualifiedSer
 		return nil, err
 	}
 
+	listenerURL, err := getListenerURL(clusterResources, cicdNamepace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get event listener URL: %v", err)
+	}
+	if accessToken == "" {
+		accessToken, err = GetAccessToken(gitRepoURL)
+		if err != nil {
+			return nil, err
+		}
+		if accessToken == "" && err == nil {
+			return nil, errors.New("unable to retrieve access-token from keyring/environment variables")
+		}
+	} else {
+		err := SetSecret(gitRepoURL, accessToken)
+		if err != nil {
+			return nil, err
+		}
+
+	}
 	repository, err := git.NewRepository(gitRepoURL, accessToken)
 	if err != nil {
 		return nil, err
 	}
+	return &webhookInfo{clusterResources, repository, gitRepoURL, cicdNamepace, listenerURL, accessToken, serviceName, isCICD}, nil
+}
 
-	listenerURL, err := getListenerURL(clusterResources, cicdNamepace)
+// GetAccessToken returns the token from either that is stored in the keyring or the environment variable in this order.
+func GetAccessToken(gitRepoURL string) (string, error) {
```

Comment:
Put this func in a new package "accesstoken" and also put the "setter in that package.   Make sure no callers are calling keyring() outside of the package.